test/inductor/test_torchinductor.py

```diff
@@ -5244,6 +5244,36 @@ def fn(x):
                 assert same(fn(x)[0], compiled([x])[0], equal_nan=True)
                 assert metrics.generated_cpp_vec_kernel_count == 1
 
+        @unittest.skipIf(
+            not codecache.valid_vec_isa_list(), "Does not support vectorization"
+        )
+        @patch("torch.cuda.is_available", lambda: False)
+        def test_vec_comapre_op_cpu_only(self):
+            def fn(x):
+                y1 = torch.eq(x, 1)
+                x = torch.where(y1, x, -x)
+                y2 = torch.ne(x, 0)
+                x = torch.where(y2, x, -x)
+                y3 = torch.lt(x, 5)
+                x = torch.where(y3, x, x - 1)
+                y4 = torch.gt(x, -2)
+                x = torch.where(y4, x, x + 1)
+                y5 = torch.le(x, 8)
+                x = torch.where(y5, x, x - 1)
+                y6 = torch.ge(x, -3)
+                x = torch.where(y6, x, x + 1)
+                return (x,)
+
+            x = torch.randn((2, 9))
+
+            with patch.object(config.cpp, "simdlen", None):
+                torch._dynamo.reset()
+                metrics.reset()
+                traced = make_fx(fn)(x)
+                compiled = compile_fx_inner(traced, [x])
+                assert same(fn(x)[0], compiled([x])[0], equal_nan=True)
+                assert metrics.generated_cpp_vec_kernel_count == 1
```

Comment:
Do we also need to assert that non vec kernel count is 0 here?

Reply:
Sure, is it fine to assert that generated_kernel_count is also 1, so that there is no non vec kernel?

Reply:
Another assert has been added: (generated_kernel_count - generated_cpp_vec_kernel_count) is 0